pages/en/lb3/Defining-middleware.md

```diff
@@ -690,17 +690,10 @@ Instead, it will be ignored because the route was already matched.
 
 ### Error-handling middleware
 
-Use error-handling middleware to deal with request errors.
+Use error-handling middleware to deal with request errors. The  [`strong-error-handler`](https://github.com/strongloop/strong-error-handler/) middleware is scaffolded by default and sufficient for many projects, but additional custom error-handling middleware may be registered as necessary.
 While you are free to register any number of error-handling middleware, be sure to register them in the "final" phase.
-LoopBack registers two error-handling middleware by default:
 
-* `urlNotFound` middleware converts all requests that reach the middleware into an Error object with status 404,
-  so that 404 error responses are consistent with "usual" error responses.
```

Comment:
You're right that this belongs somewhere. I thought it was referenced in a better place, but now see it is not. I'll bring it back into the docs, but I have a couple questions.

On investigating this again I see that there are actually two `urlNotFound` middlewares, 

  - `loopback#urlNotFound`, located at `loopback/server/middleware/url-not-found.js`. This is configured via middleware.json and is scaffolded by default.
  - `RestAdapter.urlNotFoundHandler`, located inside `strong-remoting/lib/rest-adapter.js`. This is controlled by a config.json directive, and enabled by default.

Neither of them accept any options, but they are both loaded by default in a freshly scaffolded project. The `loopback#urlNotFound` handler does not do anything because the one from strong-remoting catches everything first. 

Two questions:

1. To remain consistent with the errorHandler behavior, should we update the scaffolding and advice in docs to disable strog-remoting's `urlNotFound`, making the middleware configured `loopback#urlNotFound` the active catchall? This would make it more straightforward for an implementer to create their own catchall, by just replacing `loopback#urlNotFound` in their middleware config with their own middleware.

2. Maybe it's a mistake for there to be two of these? Is the plan to extract `urlNotFound`, as was done with strong-error-handler? Should it perhaps even be part of strong-error-handler? Any plans here?